xiangqi updated/src/Pages/SignUp.js

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import { useForm, Controller } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as Yup from 'yup';
+import Select from 'react-select';
+import 'react-toastify/dist/ReactToastify.css';
+import { useNavigate } from 'react-router-dom';
+import { toast, ToastContainer } from 'react-toastify';
+
+
+import { countryOptions, skillLevelOptions } from '../Components/options';
+import CustomDropDown from '../Components/CustomDropDown';
+import axiosInstance from "../lib/axios";
+import Button from '../Components/PlainButton';
+import Input from '../Components/Input';
+import "./Home.scss";
+
+const validationSchema = Yup.object().shape({
+    email: Yup.string()
+        .email('Invalid email address')
+        .required('Email is required'),
+    username: Yup.string()
+        .required('Username is required'),
+    password: Yup.string()
+        .min(8, 'Password must be at least 8 characters')
+        .required('Password is required'),
+    country: Yup.object().nullable()
+        .required('Country is required'),
+    skill: Yup.object().nullable()
+        .required('Skill level is required')
+});
+
+const SignUp = () => {
+    const navigate = useNavigate();
+    const { handleSubmit, control, formState: { errors } } = useForm({
+        resolver: yupResolver(validationSchema)
+    });
+
+    const onSubmit = async (data) => {
+        try {
+            const response = await axiosInstance.post('/signup', {
+                email: data.email,
+                username: data.username,
+                password: data.password,
+                country: data.country.value,
+                skill: data.skill.value
+            });
```

Comment:
As I requested earlier, we need to make a custom hook which will take url and body (if needed in case of post), and it will append headers etc. on its own!

Reply:
i'll make it in the next PR when i'll attach the django backend